### ISSUE TITLE
Better documentation for generation parameter defaults

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/transformers#12210]
It's very hard to follow how the generation parameters are set when running generation. When looking at the official function: https://github.com/huggingface/transformers/blob/700cee344691afc41f68aa18fedea463b22f95f1/src/transformers/generation_utils.py#L644 all parameters default to `None`, but are then later overwritten by the config's default parameters, *e.g.* here: https://github.com/huggingface/transformers/blob/700cee344691afc41f68aa18fedea463b22f95f1/src/transformers/generation_utils.py#L878 . This is very hard to trace or follow. We should at least put a warning or note that clearly states that all generation parameters (and actually all forward) parameters **always** default to the config.

What do you think @LysandreJik @patil-suraj @sgugger ? 

If you agree, I'll open a PR for it :-)

**Comment by sgugger** [comment]
Fine by me! I think it can just be stated at the beginning before each arg is documented.

**Comment by patrickvonplaten** [comment]
Putting this is the "Improve generation task basket" so that this is handled once the generation docs are improved as well
